core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -639,19 +640,22 @@ private[spark] object Utils extends Logging {
       fs: FileSystem,
       conf: SparkConf,
       hadoopConf: Configuration,
-      fileOverwrite: Boolean): Unit = {
-    if (!targetDir.mkdir()) {
+      fileOverwrite: Boolean,
+      filename: Option[String] = None): Unit = {
```

Comment:
Yeah. The renaming is used for the "share files among executors" feature, which creates files with a different name before moving them around. I'm not sure if that applies to directories too.